include/dpp/guild.h

```diff
@@ -382,26 +384,94 @@ class DPP_EXPORT guild_member {
 /**
  * @brief Defines a channel on a server's welcome screen
  */
-struct welcome_channel_t {
-	/// the description shown for the channel
+struct DPP_EXPORT welcome_channel: public json_interface<welcome_channel> {
+	/// The description shown for the channel
 	std::string description;
-	/// the emoji name if custom, the unicode character if standard, or null if no emoji is set
+	/// The emoji name if custom, the unicode character if standard, or null if no emoji is set
 	std::string emoji_name;
-	/// the channel's id
-	snowflake channel_id = 0;
-	/// the emoji id, if the emoji is custom
-	snowflake emoji_id = 0;
+	/// The channel's id
+	snowflake channel_id;
+	/// The emoji id, if the emoji is custom
+	snowflake emoji_id;
+
+	welcome_channel();
+
+	/**
+	 * @brief Read class values from json object
+	 *
+	 * @param j A json object to read from
+	 * @return A reference to self
+	 */
+	welcome_channel& fill_from_json(nlohmann::json* j);
+
+	/**
+	 * @brief Build the json for this object
+	 *
+	 * @param with_id include the id in the JSON
+	 * @return std::string json data
+	 */
+	std::string build_json(bool with_id = false) const;
+
+	/**
+	 * @brief Set the channel ID of this welcome channel object
+	 *
+	 * @param _channel_id The channel ID to set
+	 * @return Reference to self, so these method calls may be chained
+	 */
+	welcome_channel& set_channel_id(const snowflake _channel_id);
+
+	/**
+	 * @brief Set the description of this welcome channel object
+	 *
+	 * @param _description The description to set
+	 * @return Reference to self, so these method calls may be chained
+	 */
+	welcome_channel& set_description(const std::string& _description);
 };
 
 
 /**
  * @brief Defines a server's welcome screen
  */
-struct welcome_screen_t {
-	/// the server description shown in the welcome screen
+struct DPP_EXPORT welcome_screen: public json_interface<welcome_screen> {
+	/// The server description shown in the welcome screen
 	std::string description;
-	/// the channels shown in the welcome screen, up to 5
-	std::vector<welcome_channel_t> welcome_channels;
+	/// The channels shown in the welcome screen (max 5)
+	std::vector<welcome_channel> welcome_channels;
+	/// Whether the welcome screen should be enabled or disabled. It's empty for received welcome-screens. Only for modifying the welcome screen
+	std::optional<bool> enabled;
```

Comment:
Not sure if this is the right place. It's only used when editing the welcome screen: https://discord.com/developers/docs/resources/guild#modify-guild-welcome-screen. It's stored in the guild class, so it would take more ram. I'm thinking if i should make this an extra parameter of `cluster::guild_edit_welcome_screen`. wdym?

Reply:
i think youre right, this is an excessive amount of extra storage, for something like the welcome channels which the majority of servers simply wont have. Also does it need an enabled flag to be optional? that seems like a tri-state bool which is just ugh. its enabled, or it isnt.

Reply:
optional because it's an optional field in the endpoint

Reply:
should i make two different structs of welcome screens? one `welcome_screen_t` as it was before, and `welcome_screen` which is derived from `welcome_screen_t` and has this extra boolean and is only used for the two welcome-screen-endpoints?

Reply:
hm nah, sounds too complicated i guess. ill add the bool to `guild_edit_welcome_screen`